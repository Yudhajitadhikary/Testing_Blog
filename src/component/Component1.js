
import React,{useState} from 'react';

function Component1(props) {
    const [authors, setAuthors] = useState([]);
    //    const [activeAuthor, setActiveAuthor] = useState(null);
         const [posts, setPosts] = useState([]);
         //     // Load authors on start
    React.useEffect(() => {
      setAuthors([{id:1,name:'Yudhajit Adhikary',post:'Blog'},{id:1,name:'Koyena Adhikary',post:'Blog1'}])
      setPosts([{id:1,name:'Yudhajit Adhikary',post:'Blog'},{id:1,name:'Koyena Adhikary',post:'Blog1'}])
     },[]);
  
   
      
    return (
              <div style={{display:'flex'}}>
              <div>{authors&&<div>{authors.map((author,index)=>(<div>{author.name}</div>))}</div>}</div>
              <div>{posts&&<div>{posts.map((post,index)=>(<div>{post.post}</div>))}</div>}</div>
              </div>
              
            );
    }
    export default Component1;      

