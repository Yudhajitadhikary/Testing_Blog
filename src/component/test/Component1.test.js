import React from "react";
import { mount } from "../enzyme";
import { shallowToJson } from "enzyme-to-json";
import Component1 from '../Component1';
import { Provider } from 'react-redux';
import store from '../../store';
 
let wrapper;
it("should render TextFieldAtom component", () => {
  wrapper = mount(<Provider store={store}><Component1/></Provider>);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
  

});

