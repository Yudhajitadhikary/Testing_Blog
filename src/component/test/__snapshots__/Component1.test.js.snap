// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextFieldAtom component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Component1>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div>
        <div>
          <div>
            Yudhajit Adhikary
          </div>
          <div>
            Koyena Adhikary
          </div>
        </div>
      </div>
      <div>
        <div>
          <div>
            Blog
          </div>
          <div>
            Blog1
          </div>
        </div>
      </div>
    </div>
  </Component1>
</Provider>
`;
